### PR TITLE
Customer registration, login, forgotpassword

### PR DESCRIPTION
**Pull request by vamcart**
Customer registration, login, password recovery!
#63

### CODE REVIEW

**Comment by vamcart** [comment]
And here is the my working copy of this changes.
All works fine for me.
https://cezerin2.slack.com/files/UFVAW9HA4/FGCL59ETV/cezerin-customer-register-login-account.zip

**Comment by vamcart** [comment]
To test new email templates, features, i have local smtp server.
Setup your smtp server at /config/server.js to test account activation, password recovery feature.

New email templates and options added to the "npm run setup"
/src/api/server/setup.js

**Comment by vamcart** [comment]
If you have error when you click register button, try to clear cache in Chrome - Developer Console - Application - Clear storage - Clear site data.

actions.js:774 Uncaught (in promise) TypeError: Cannot read property 'retrieve' of undefined
in /src/store/shared/actions.js
:
export const registerUser = (data, callback) => async (dispatch, getState) => {
    const response = await api.ajax.register.retrieve(data);
    dispatch(handleRegisterProperties(response.json));
};
this string:
api.ajax.register.retrieve(data);

**Comment by LuBuss** [comment]
I have partly reviewed, API seems to be working correctly:

- passwords are stored as plain strings (encrypt and hash these.)
- The customer is currently required to create an account, (I feel if choose guest we should not make them.)

**Comment by LuBuss** [comment]
I can verify all working, have not yet reviewed code

**Comment by vamcart** [comment]
> I have partly reviewed, API seems to be working correctly:
> 
> * The customer is currently required to create an account, (I feel if choose guest we should not make them.)

I don't think so. Leave as is now.

File: src/lib/security.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ const applyMiddleware = app => {
 };
 
 const getAccessControlAllowOrigin = () => {
-	return [settings.storeBaseUrl, settings.adminBaseURL] || '*';
+	return settings.storeBaseUrl || '*';
```
Comment (ngoyal16) [inline src/lib/security.js]: add admin URL into it

File: src/services/settings/settings.js [hunk 2]
```diff
@@ -11,8 +11,9 @@ import parse from '../../lib/parse';
 class SettingsService {
 	constructor() {
 		this.defaultSettings = {
+			store_name: 'Cezerin',
 			domain: '',
-			logo_file: null,
+			logo_file: 'logo.png',
```
Comment (ngoyal16) [inline src/services/settings/settings.js:16]: the default logo should be part of the theme.

**Review by ngoyal16 — DISMISSED** [review]

**Comment by vamcart** [comment]
When this PR will be merged?!

I test it on my local machine and it's ok.

New version of cezerin2-client - ok.
Customer Registration - ok.
Customer Activation - ok.
Customer Order Checkout - ok.
Customer Logoff - ok.
Customer Account Page - ok.
Customer Edit Address Form - ok.
New settings for "npm run setup" - ok.

All of this works fine at my local machine.

File: src/lib/security.js (hunk 1 above)
Comment (LuBuss) [reply src/lib/security.js]: @vamcart adminBaseURL needs to be readded or admin will not work

Comment (LuBuss) [reply src/lib/security.js]: Once this is fixed I can merge it in.

Comment (vamcart) [reply src/lib/security.js]: > Once this is fixed I can merge it in.

ok. string is back.
here is the commit - https://github.com/Cezerin2/cezerin2/pull/64/commits/b2bfaf74ede38d0136972412fde42ab3bee64e22

I'm found issue in cezerin2-client, problem with saving customer data when i editing customer addresses at customer page.

This is the fix fox that issue - https://github.com/Cezerin2/cezerin2-client/pull/11

**Review by LuBuss — APPROVED** [review]

**Pull request merged**